agent/tags.go

```diff
@@ -157,14 +157,24 @@ func (t *tagFetcher) Fetch(l logger.Logger, conf FetchTagsConfig) []string {
 
 	// Attempt to add the default GCP meta-data tags
 	if conf.TagsFromGCPMetaData {
-		gcpTags, err := t.gcpMetaDataDefault()
-		if err != nil {
-			// Don't blow up if we can't find them, just show a nasty error.
-			l.Error(fmt.Sprintf("Failed to fetch Google Cloud meta-data: %s", err.Error()))
-		} else {
-			for tag, value := range gcpTags {
-				tags = append(tags, fmt.Sprintf("%s=%s", tag, value))
+		l.Info("Fetching GCP meta-data...")
+
+		err := retry.Do(func(s *retry.Stats) error {
+			gcpTags, err := t.gcpMetaDataDefault()
+			if err != nil {
+				// Don't blow up if we can't find them, just show a nasty error.
+				l.Error(fmt.Sprintf("Failed to fetch GCP meta-data: %s", err.Error()))
+			} else {
+				for tag, value := range gcpTags {
+					tags = append(tags, fmt.Sprintf("%s=%s", tag, value))
+				}
 			}
+			return err
+		}, &retry.Config{Maximum: 5, Interval: 1 * time.Second, Jitter: true})
```

Comment:
I wonder if five seconds will always be enough for the described case of network not being online yet.
I haven't looked into the retry code, but do we have a back-off option? Or maybe just crank up the retry count?